.github/instructions/prompt-tsx.instructions.md

```diff
@@ -0,0 +1,166 @@
+---
+applyTo: '**/*.tsx'
+description: Prompt-TSX coding guidelines
+---
+
+Guidelines for TSX files using [prompt-tsx](https://github.com/microsoft/vscode-prompt-tsx) focusing on specific patterns and token budget management for AI prompt engineering.
+
+## Component Structure
+
+### Base Pattern
+- Extend `PromptElement<Props>` or `PromptElement<Props, State>` for all prompt components
+- Props interfaces must extend `BasePromptElementProps`
+
+```tsx
+interface MyPromptProps extends BasePromptElementProps {
+	readonly userQuery: string;
+}
+
+class MyPrompt extends PromptElement<MyPromptProps> {
+	render() {
+		return (
+			<>
+				<SystemMessage priority={1000}>...</SystemMessage>
+				<UserMessage priority={900}>{this.props.userQuery}</UserMessage>
+			</>
+		);
+	}
+}
+```
+
+### Async Components
+- Use `prepare` method for async operations before rendering
+- Pass prepared state to `render` method
+
+```tsx
+class FileContextPrompt extends PromptElement<Props, PreparedState> {
+	override async prepare(sizing: PromptSizing): Promise<PreparedState> {
+		// Async preparation logic
+		return state;
+	}
```

Comment:
I suggest removing this, `render` is async now so there is not much reason at all to use prepare